source/blog/2014-02-04-ember-1-10-0-released.md

```diff
@@ -0,0 +1,387 @@
+    Hello, {{name}}. You are {{age}} years old.
+  {{/x-customer}}
+</div>
+```
+
+Many thanks to [@\_mmun](https://twitter.com/_mmun) for the implementation of this new feature.
+
+#### Injected Properties
+
+Ember 1.x has exposed two APIs for managing depenency injection. The first is
+the application initializer API, using `register` and `inject` methods on an
+application instance. The second allows configuration of an injection on the
+class itself via `needs`. You can read more about these patterns in
+[the dependency injection guide](http://emberjs.com/guides/understanding-ember/dependency-injection-and-service-lookup/).
+
+The new injected properties offer a more declarative API for dependency injection.
+
+Use `Ember.inject.service()` to inject a service with the same name as the property
+it is injected as. In this example, the `storage` service is injected onto the
+`storage` property:
+
+```javascript
+export default Ember.Component.extend({
+  storage: Ember.inject.service()
+});
+```
+
+Passing a name to the `service()` method allows a different service to be injected.
+For example:
+
+```javascript
+export default Ember.Component.extend({
+  storage: Ember.inject.service('local-storage')
+});
+```
+
+In addition to injecting services, controllers can be injected with `Ember.inject.controller`.
+
+Importantly, these injections are lazy. When used within a unit test, a stubbed service can
+be passed during `create`. For example:
+
+```javascript
+test("a value is saved on storage", function(){
+  expect(1);
+  var component = this.subject({
+    storage: {
+      write: function(){
+        ok(true, 'value was written');
+      }
+    }
+  });
+
+  var element = this.append();
+  element.click();
+});
+```
+
+Refer to the [API documentation](add this) to read about this new feature in
+detail.
+
+Thanks to [slindberg](https://github.com/slindberg) for his implementation of this feature.
+
+#### Notable Deprecations
+
+As Ember.js moves forward, various APIs are deprecated to allow for their
+removal in a later major release (such as 2.0). The
+[deprecations guide](http://emberjs.com/guides/deprecations/) summarizes
+deprecations and demonstrates how to update to a new API.
+
+* The explicit `{{bind` helper has been deprecated. This helper has
+  long been marked private, and was a legacy Sproutcore
+  feature. This helper will be removed in Ember 1.11.
+* Quote-less outlet names are deprecated in 1.10. An example of this is
+  `{{outlet modal}}`, which should be re-written as `{{outlet "modal"}}`.
+  This ensures the outlet helper is consistent with others, where unquoted
+  words are values and not string literals.
+* The `beforeObserver` feature is deprecated in Ember 1.10. Before observers
+  are rarely used, but introduce significant overhead to the observer system
+  in general. For observer use that requires the previous value of a property
+  be known, implementing a cache is simple and more efficient. Read more about
```

Comment:
Ah, yeah. I read this as a parse error but it just has a meaning I was not expecting. Carry on.